src/scripts/validate_synapse_records/validate_synapse_records.py

```diff
@@ -0,0 +1,172 @@
+"""
+This script validates record data against its schema by querying
+a Synapse file view for records.
+
+The records must use the self-referencing format in their metadata.json file.
+Any files in the record which do not have a `jsonSchema` property specified in
+their fileInfo object in metadata.json will not be validated.
+
+The validation results are written to `validation_errors.json`.
+"""
+
+import json
+import zipfile
+import argparse
+import boto3
+import jsonschema
+import requests
+import synapseclient
+
+def read_args():
+    parser = argparse.ArgumentParser(
+            description="Validate records in a Synapse file view against their JSON Schema.")
+    parser.add_argument("--file-view",
```

Comment:
Just out of curiosity, since positional arguments are required, why not make this a positional argument?

Reply:
Why do you say that positional arguments are required? I find positional arguments to be relatively ambiguous and increase the likelihood of passing incorrect arguments, so I tend to always use flags.

Reply:
Hmmm, I misspoke, it seems like all arguments by default is None.   I agree with your usage of flags